### PR TITLE
UIMARCAUTH-208: Global Heading Report | UX | MARC authority headings updates report

### PR DESCRIPTION
**Pull request by BogdanDenis**
## Description
Create headings updates report modal

**Note: BE implementation not ye done, so http request returns an error**

## Screenshots

https://user-images.githubusercontent.com/19309423/215126419-ee6176ac-0328-4050-8fce-d83a77fa1afe.mp4

## Issues
[UIMARCAUTH-208](https://issues.folio.org/browse/UIMARCAUTH-208)

## Related PRs
https://github.com/folio-org/stripes-components/pull/1968

### CODE REVIEW

File: src/views/AuthoritiesSearch/ReportsModal/ReportsModal.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+import { REPORT_TYPES } from '../constants';
+
+import styles from './ReportsModal.css';
+
+const propTypes = {
+  handleSubmit: PropTypes.func.isRequired,
+  onClose: PropTypes.func.isRequired,
+  open: PropTypes.bool.isRequired,
+  reportType: PropTypes.oneOf(Object.values(REPORT_TYPES)).isRequired,
```
Comment (Dmytro-Melnyshyn) [inline src/views/AuthoritiesSearch/ReportsModal/ReportsModal.js:25]: This is not required, the component is rendered even if `open` is false.

Comment (BogdanDenis) [reply src/views/AuthoritiesSearch/ReportsModal/ReportsModal.js:25]: The component is rendered, but it returns `null` if `open` is false

File: src/views/AuthoritiesSearch/AuthoritiesSearch.js [hunk 2]
```diff
@@ -372,13 +398,13 @@ const AuthoritiesSearch = ({
...
 
-  const sortByOptions = [
+  const sortByOptions = useMemo(() => [
     {
       value: '',
       label: intl.formatMessage({ id: 'ui-marc-authorities.actions.menuSection.sortBy.relevance' }),
     },
     ...options,
-  ];
+  ], [intl, options]);
```
[the remark this answers is not in the document]
Comment (BogdanDenis) [reply src/views/AuthoritiesSearch/AuthoritiesSearch.js:407]: used `useMemo` for `options` too

File: src/views/AuthoritiesSearch/ReportsModal/ReportsModal.js (hunk 1 above)
Comment (Dmytro-Melnyshyn) [reply src/views/AuthoritiesSearch/ReportsModal/ReportsModal.js:25]: With null, we'll see the warning like `reportType is marked as required, but its value is null`

**Review by Dmytro-Melnyshyn — APPROVED** [review]

**Review by Dmytro-Melnyshyn — APPROVED** [review]

File: src/views/AuthoritiesSearch/ReportsModal/ReportsModal.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        }) => (
+          <div className={styles.dateRangeWrapper}>
+            <Field
+              name="fromDate"
+              label={intl.formatMessage({ id: 'ui-marc-authorities.reportModal.startDate' })}
+              required
+              component={Datepicker}
+              exclude={startDateExclude}
+              usePortal
+            />
```
Comment (Dmytro-Melnyshyn) [inline src/views/AuthoritiesSearch/ReportsModal/ReportsModal.js:82]: I think it is needed to add validation for the fields.
validate={value => value ? undefined : 'Error'}

Comment (BogdanDenis) [reply src/views/AuthoritiesSearch/ReportsModal/ReportsModal.js:84]: Added validation for the form

**Pull request merged**